app/importers/importers/importer.rb

```diff
@@ -22,63 +22,104 @@ class Importer
     #
     # Argument metadata is a hash read from an individual import.json file, contents
     # differ depending on file type.
+    # options:
+    # keep_conflicting_items: default true, if false will delete conflicting items to make room for incoming items.
     def initialize(metadata, options = {})
-      #raise ArgumentError unless target_item.is_a? self.class.exportee
       @metadata = metadata
       @errors ||= []
+      @conflicting_item_was_kept = false
+      explanation = 'options[:keep_conflicting_items] should be true, false or nil.'
+      raise ArgumentError, explanation unless [true,false,nil].include? options[:keep_conflicting_items]
+      @keep_conflicting_items = options[:keep_conflicting_items].nil? ? true : options[:keep_conflicting_items]
+    end
+
+    # This is the only method called on this class
+    # by the rake task after instantiation.
+    # It reads metadata from file, creates or finds
+    # an item based on it, blanks out some of its metadata for freshening up from import if needed,
+    # then saves it to the database.
+    def import
+      self.class.without_auto_timestamps do
+        if preexisting_item?
+          blank_out_for_reimport(target_item)
+        end
+        return if @conflicting_item_was_kept
+        common_populate
+        populate
+        save_target_item
+      end
+    end
+
+    # Is there a preexisting item with the same friendlier_id *and* the same type?
+    def preexisting_item?
+      # Calling target_item:
+      #   * checks for any conflicting items
+      #       * deletes them if it is allowed to,
+      #       * sets @conflicting_item_was_kept otherwise.
+      #   * creates or sets @target_item (without blanking it out -- that's done in the import method.)
+      #   * sets boolean ivar @preexisting_item_was_found
+      target_item
```

Comment:
This all seems overly complex and indirect, that you need to document here what some _other_ method does seems a sign that the given methods are not responsible for clear single things. It makes the code very hard to follow. It is also better design when methods return as return values the things they are supposed to return, instead of setting ivars as side-effects -- makes it hard to keep track of where the ivars are being set (could be multiple places), and who consumes them (could be multiple places again). 

All this code just seems overly complex. But it's not obvious to me what the solution is, and it's not clear if it's worth it for the import script. I'm not really sure what to do.